File: src/main/java/com/example/common/entity/Scheme.java
package com.example.common.entity;


import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;
import org.hibernate.search.annotations.Field;
import org.hibernate.search.annotations.Indexed;
import org.hibernate.search.annotations.IndexedEmbedded;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

/**
 * Created by Anatol on 11.06.16.
 */
@Entity
@Table(name = "scheme")
@Indexed
public class Scheme extends BaseEntity {

    @Column(nullable = false)
    @Field
    private String name;

    @Column(nullable = false)
    @Field
    private String description;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    @Fetch(FetchMode.SELECT)
    private User user;

    @Column(nullable = false)
    private Long creationDate;

    @Column(nullable = false)
    @Field
    private String category;

    @OneToMany(mappedBy = "schemeId", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @Fetch(FetchMode.SELECT)
    private List<SchemeRating> rates;

    @OneToMany(mappedBy = "schemeId", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @Fetch(FetchMode.SELECT)
    private List<Node> nodes;

    @OneToMany(mappedBy = "schemeId", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @Fetch(FetchMode.SELECT)
    private List<Line> lines;

    @OneToMany(mappedBy = "schemeId", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @Fetch(FetchMode.SELECT)
    private List<ElementCoordinates> elements;

    @ManyToMany(cascade= CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinTable(name="tag_scheme", joinColumns=@JoinColumn(name="scheme_id"),
            inverseJoinColumns=@JoinColumn(name="tag_id"))
    @IndexedEmbedded
    private Set<Tag> tags;

    public List<Node> getNodes() {
        return nodes;
    }

    public void setNodes(List<Node> nodes) {
        this.nodes = nodes;
    }

    public List<Line> getLines() {
        return lines;
    }

    public void setLines(List<Line> lines) {
        this.lines = lines;
    }

    public Set<Tag> getTags() {
        return tags;
    }

    public void setTags(Set<Tag> tags) {
        this.tags = tags;
    }

    public List<ElementCoordinates> getElements() {
        return elements;
    }

    public void setElements(List<ElementCoordinates> elements) {
        this.elements = elements;
    }

    public List<SchemeRating> getRates() {
        return rates;
    }

    public void setRates(List<SchemeRating> rates) {
        this.rates = rates;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Long getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Long creationDate) {
        this.creationDate = creationDate;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

}

File: frontend/src/app/dto/Scheme.ts
import User from "./User";
import SchemeRating from "./SchemeRating";
import ElementCoordinates from "./ElementCoordinates";
import Tag from "./Tag";
import Line from "./Line";
import Node from "./Node";
export default class Scheme {
    private id: number;
    private user: User;
    private name: string;
    private description: string;
    private category: string;
    private creationDate: number;
    private rates: SchemeRating[];
    private elements:Array<ElementCoordinates>;
    private tags:  Array<Tag>;
    private lines: Array<Line>;
    private nodes: Array<Node>;

  constructor(id:number, user:User, name:string, description:string, category:string, creationDate:number,
              rates:SchemeRating[], elements:Array<ElementCoordinates>, tags:Array<Tag>, lines:Array<Line>,
              nodes:Array<Node>) {
    this.id = id;
    this.user = user;
    this.name = name;
    this.description = description;
    this.category = category;
    this.creationDate = creationDate;
    this.rates = rates;
    this.elements = elements;
    this.tags = tags;
    this.lines = lines;
    this.nodes = nodes;
  }

  getId():number{
      return this.id;
      }

  setId(value:number){
      this.id=value;
      }

  getUser():User{
      return this.user;
      }

  setUser(value:User){
      this.user=value;
      }

  getName():string{
      return this.name;
      }

  setName(value:string){
      this.name=value;
      }

  getDescription():string{
      return this.description;
      }

  setDescription(value:string){
      this.description=value;
      }

  getCategory():string{
      return this.category;
      }

  setCategory(value:string){
      this.category=value;
      }

  getCreationDate():number{
      return this.creationDate;
      }

  setCreationDate(value:number){
      this.creationDate=value;
      }

  getRates():SchemeRating[]{
      return this.rates;
      }

  setRates(value:Array){
      this.rates=value;
      }

  getElements():Array<ElementCoordinates>{
      return this.elements;
      }

  setElements(value:Array<ElementCoordinates>){
      this.elements=value;
      }

  getTags():Array<Tag>{
      return this.tags;
      }

  setTags(value:Array<Tag>){
      this.tags=value;
      }

  getLines():Array<Line>{
      return this.lines;
      }

  setLines(value:Array<Line>){
      this.lines=value;
      }

  getNodes():Array<Node>{
      return this.nodes;
      }

  setNodes(value:Array<Node>){
      this.nodes=value;
      }
}
